### PR TITLE
feat(core): implement vertical navigation

### PR DESCRIPTION
**Pull request by mikerodonnell89**
## Related Issue.
Closes SAP/fundamental-ngx/#6328

## Description
Implements the vertical navigation component in fundamental-ngx/core.

## Screenshots
Screen Shot 2021-09-23 at 11 49 48 AM
Screen Shot 2021-09-23 at 11 49 51 AM
Screen Shot 2021-09-23 at 11 50 00 AM
Screen Shot 2021-09-23 at 11 50 05 AM

### CODE REVIEW

File: libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        </ul>
+    </div>
+    <div
+        class="fd-popover__body fd-popover__body--no-arrow fd-list__navigation-item-popover--second-level"
+        aria-hidden="false"
+    >
+        <ul class="fd-list">
+            <li
+                *ngFor="let navItem of _listComponent._navItems"
+                [ngClass]="{ 'fd-list__navigation-item--indicated': navItem.indicated }"
```
Comment (Betrozov) [inline libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html:24]:
```suggestion
                [сlass.fd-list__navigation-item--indicated]="navItem.indicated"
```

File: libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                class="fd-list__navigation-item fd-list__navigation-item--condensed"
+                tabindex="0"
+            >
+                <span class="fd-list__navigation-item-text">{{ navItem?._innerText }}</span>
```
Comment (Betrozov) [inline libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html:28]: do we need empty span if navItem does not exist?

File: libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        if (this._iconComponent) {
+            this._iconComponent._navigationItemIcon = true;
+        }
+        this._innerText = this._text.elementRef.nativeElement.innerText;
+    }
+
+    /** @hidden */
+    @HostListener('click', ['$event'])
+    onItemClick(event): void {
```
Comment (Betrozov) [inline libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.ts:74]: please add type

File: libs/core/src/lib/vertical-navigation/vertical-navigation.component.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div class="fd-vertical-nav" [ngClass]="{ 'fd-vertical-nav--condensed': condensed }">
```
Comment (Betrozov) [inline libs/core/src/lib/vertical-navigation/vertical-navigation.component.html:1]:
```suggestion
<div class="fd-vertical-nav" [class.fd-vertical-nav--condensed]="condensed">
```

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Review by Smiranin — APPROVED** [review]

File: libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html (hunk 1 above)
Comment (mikerodonnell89) [reply libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html:24]: Why? My IDE doesn't like this syntax
Screen Shot 2021-09-27 at 10 49 49 AM

File: libs/core/src/lib/vertical-navigation/vertical-navigation.component.html (hunk 4 above)
Comment (mikerodonnell89) [reply libs/core/src/lib/vertical-navigation/vertical-navigation.component.html:1]: See above comment

File: libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html (hunk 1 above)
Comment (Betrozov) [reply libs/core/src/lib/list/list-navigation-item/list-navigation-item.component.html:24]: `[class]` preferred over the `[ngClass]`
https://angular.io/guide/built-in-directives#adding-and-removing-classes-with-ngclass
https://angular.io/guide/attribute-binding#binding-to-multiple-css-classes

But it behaves something differently:
https://github.com/angular/angular/issues/40623
https://hackmd.io/jzDc7hIDTdWtQblv2TbL9A

**Review by Betrozov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**